### ISSUE TITLE
[css-ui] Should mention the posibility to ignore custom cursors that go outside of the viewport

### ISSUE THREAD

**Issue by emilio** [w3c/csswg-drafts#3728]
Both Firefox and Blink are switching to limit custom cursors bigger than a particular size which intersect UI:

 * https://bugzilla.mozilla.org/show_bug.cgi?id=1445844
 * https://bugs.chromium.org/p/chromium/issues/detail?id=880863

It was suggested in the dev-platform@ intent to ship thread that maybe adding some kind of text to the spec for this case could be worth it.

**Comment by frivoal** [comment]
Right. Given that this limitation is not rooted in picking some arbitrary limit for implementation convenience, but rather finding it desirable that cursors larger than a certain size get ignored (in some cases) for security reasons, reflecting that in the spec would seem reasonable to me.

**Comment by frivoal** [comment]
If I understand correctly, both Firefox and Chrome have now shipped some variant of this, but what you have shipped isn't exactly the same. I'm curious if we learned anything after shipping, which should influence what goes in the spec.

As a reminder, what the spec currently has to say about cursors that are too large is:
> 	If an operating system is incapable of rendering a cursor above a given size, cursors larger than that size must be shrunk to within the OS-supported size bounds, while maintaining the cursor image’s intrinsic ratio, if any.

Maybe we want to keep this as is and add a "may reject or clip cursors above a certain size, for security reasons", but if nobody does the shrinking, we way want to do away with that.

Agenda+ to raise attention and attempt to get feedback.

**Comment by emilio** [comment]
What is the "not exactly the same" part, other than the value being configurable in Gecko? IIUC both engines should fall back to the next cursor.

**Comment by smfr** [comment]
I'm confused. Is this issue about cursor size, or about custom cursors that have hotpoints configured such that they try to confuse the user about where they are clicking?

**Comment by emilio** [comment]
@smfr the combination of both. Basically, if you have a big cursor that due to the hotspot would go outside of the content viewport.

**Comment by smfr** [comment]
I think a diagram of the potential attack here would be useful, but I get the idea.

**Comment by emilio** [comment]
You can basically trick the user so that they think they are pointing to the URL bar but instead they're pointing inside the page.

You can see a (very intentionally obvious) test-case in https://bug1445844.bmoattachments.org/attachment.cgi?id=9030715. You can disable `layout.cursor.block.enabled` in Firefox's `about:config` to see the before / after.

**Comment by drecali** [comment]
If anyone wants to see an interactive demo of this, the Bugzilla link above has an attachment that's shows how a bad actor can trick a user about where they're actually pointing. Here it is: https://bug1445844.bmoattachments.org/attachment.cgi?id=9030715

@emilio does any further action need to be taken about this issue. If so, can I help with it? I checked https://www.w3.org/TR/css-ui-4/ and https://www.w3.org/TR/css-ui-3/ and I didn't notice anything addressing this issue. I did see it in the MDN docs, but the 32px limit is not explicitly mentioned (maybe because it might change in the future?)
